src/Control/Distributed/Spark/SQL/StructType.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE DataKinds #-}
```

Comment:
I feel it would be better to split this module into one for `StructType`, one for `StructField` and one for `Metadata`, and design the API of each to be imported qualified, sticking to the upstream names wherever possible. Otherwise it becomes hard to find a binding.